src/Lean/Elab/BuiltinCommand.lean

```diff
@@ -223,6 +341,42 @@ private def replaceBinderAnnotation (binder : TSyntax ``Parser.Term.bracketedBin
   else
     return #[binder]
 
+/-- Declares a list of typed variables.
```

Comment:
I think that this docstring is also missing a motivation for why someone would want to use the feature. After all, one might think, Lean already generalized unbound names to implicit arguments! What gives?

I believe the answer is:
 * Lean doesn't always pick the correct order of the arguments, but that can matter in the presence of dependent types
 * Lean can't always infer the type, and it can be nicer to write it once rather than in many related definitions/declarations
 * Lean won't always add enough dependencies when complex dependent types are involved
 * Being explicit about variable types improves error messages
 * This feature allows overriding a globally-defined name to use it as an implicit argument again in the section

Did I miss any others?

Are there other use cases?

Reply:
Well I would say the main use case is to make (type) definitions much easier to write when you're writing a bunch of things that have roughly the same signature, especially when you have typeclass parameters all over the place.

Reply:
This would be a great second sentence in your docstring :-)

Reply:
Added, along with an example as I was worried it was not understandable unless you already know what it means

Reply:
I'm a bit worried the example is too long for what it conveys, I'm interested in feedback